packages/workspace/src/workspace/nacl_files/nacl_files_source.ts

```diff
@@ -454,12 +454,24 @@ const buildNaclFilesSource = (
         })
         .value(),
       DUMP_CONCURRENCY
-    )).filter(b => b !== undefined) as Required<ParsedNaclFile>[]
+    )).filter(values.isDefined)
 
     if (updatedNaclFiles.length > 0) {
       log.debug('going to update %d NaCl files', updatedNaclFiles.length)
-      await setNaclFiles(...updatedNaclFiles)
-      const res = await buildNaclFilesStateInner(updatedNaclFiles)
+      // The map is to avoid saving unnecessary fields in the nacl files
+      await setNaclFiles(...updatedNaclFiles.map(naclFile => ({
+        buffer: naclFile.buffer,
+        timestamp: naclFile.timestamp,
+        filename: naclFile.filename,
+      })))
+      // The map is to avoid saving unnecessary fields in the state
+      const res = await buildNaclFilesStateInner(updatedNaclFiles.map(parsedNaclFile => ({
+        filename: parsedNaclFile.filename,
+        elements: parsedNaclFile.elements,
+        errors: parsedNaclFile.errors,
+        timestamp: parsedNaclFile.timestamp,
+        referenced: parsedNaclFile.referenced,
+      })))
```

Comment:
a little shorter implementation that still seems to keep type safety:
```suggestion
      // The map is to avoid saving unnecessary fields in the nacl files
      await setNaclFiles(
        ...updatedNaclFiles.map(file => _.pick(file, ['buffer', 'filename', 'timestamp']))
      )
      const res = await buildNaclFilesStateInner(
        updatedNaclFiles.map(file => _.omit(file, 'buffer'))
      )
```

Reply:
using omit seems kinda dangerous seems if another field will be added to NaclFile, we won't know and pass it to the state, so I used pick  as in line 463 instead

Reply:
just depends on whether the added field is optional or required.
if it is optional then you are right, if it is required then using `pick` will break the function call

not super happy that we have to basically define the field names of the type in to decoupled locations but both options are not ideal in that regard so I'm ok with either